### PR TITLE
Hpe msa 2022

### PR DESCRIPTION
**Pull request by luojiagen**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 1]
```diff
@@ -706,6 +691,36 @@ def list_port_groups(self, storage_id):
...
             raise e
 
+    @staticmethod
+    def get_port_group_id_and_name(port_number, storage_port_list):
+        native_port_group_id = None
+        native_port_group_name = None
+        if port_number:
+            port_codes = port_number.split(',')
+            for port_code in port_codes:
+                if storage_port_list:
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:701]: if 'storage_port_list' is a list, no need to add 'if storage_port_list'

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 2]
```diff
@@ -746,47 +765,56 @@ def get_storage_view_list(storage_view_list, vol_type, storage_id,
...
                         view_map['native_port_group_id'] = \
                             "port_group_" + native_port_group_id
                     views_list.append(view_map)
             return views_list
+
+    @staticmethod
+    def get_storage_host_id(host_list, mapped_id, initiators_list,
+                            storage_host_groups, view_name):
+        native_storage_host_id = None
+        if host_list:
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:798]: if 'host_list' is a list, no need to add 'if host_list'

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 3]
```diff
@@ -746,47 +765,56 @@ def get_storage_view_list(storage_view_list, vol_type, storage_id,
...
+            for host_value in host_list:
+                host_durable_id = host_value.get('description')
+                if host_durable_id == mapped_id:
+                    native_storage_host_id = host_value.get(
+                        'native_storage_host_id')
+        if not native_storage_host_id:
+            if initiators_list:
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:805]: if 'initiators_list' is a list, no need to add 'if initiators_list'

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 4]
```diff
@@ -746,47 +765,56 @@ def get_storage_view_list(storage_view_list, vol_type, storage_id,
...
+                for initiators in initiators_list:
+                    initiators_durable_id = initiators.get(
+                        'native_storage_host_initiator_id')
+                    if initiators_durable_id == mapped_id:
+                        native_storage_host_id = initiators.get(
+                            'native_storage_host_id')
+        if not native_storage_host_id:
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:812]: Please put these two judgments('if not native_storage_host_id') together

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 5]
```diff
@@ -560,20 +560,21 @@ def list_storage_host_groups(self, storage_id):
...
-                                    'storage_id': storage_id,
-                                    'native_storage_host_group_id':
-                                        storage_host_group_id,
-                                    'native_storage_host_id': host_id
-                                }
-                                storage_host_grp_relation_list.\
-                                    append(storage_host_group_relation)
+                            if host_id != 'NOHOST':
+                                if host_group_id == storage_host_group_id:
+                                    if hosts_str:
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:565]: Please simplify the code, for example:
You can define 'hosts_str' as list type, last use ' ','join(hosts_str) '.
This also achieves the same result.

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py (hunk 1 above)
Comment (luojiagen) [reply delfin/drivers/hpe/hpe_msa/ssh_handler.py:701]: Delete judgment if 'storage_port_list'

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py (hunk 2 above)
Comment (luojiagen) [reply delfin/drivers/hpe/hpe_msa/ssh_handler.py:798]: Delete judgment if 'storage_port_list'

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py (hunk 3 above)
Comment (luojiagen) [reply delfin/drivers/hpe/hpe_msa/ssh_handler.py:805]: Delete judgment if 'initiators_list'

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py (hunk 2 above)
Comment (luojiagen) [reply delfin/drivers/hpe/hpe_msa/ssh_handler.py:798]: Delete judgment if 'host_list'

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py (hunk 5 above)
Comment (luojiagen) [reply delfin/drivers/hpe/hpe_msa/ssh_handler.py:565]: The code has been optimized

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 6]
```diff
@@ -554,32 +554,29 @@ def list_storage_host_groups(self, storage_id):
             host_info_list = self.handle_xml_to_json(host_groups_info, 'host')
             if host_groups:
                 for host_group in host_groups:
-                    hosts_str = None
+                    hosts_str = []
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:557]: 'hosts_str' is a list, variable name and type do not match

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 7]
```diff
@@ -604,18 +601,14 @@ def list_volume_groups(self, storage_id):
                 volume_groups_info, 'volumes')
             if volume_groups_json:
                 for volume_group in volume_groups_json:
-                    volumes_str = None
+                    volumes_str = []
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:604]: Variable name and type do not match

File: delfin/drivers/hpe/hpe_msa/ssh_handler.py [hunk 8]
```diff
@@ -746,47 +746,53 @@ def get_storage_view_list(storage_view_list, vol_type, storage_id,
...
-                                    'native_storage_host_id')
-                    if not native_storage_host_id:
-                        if initiators_list:
-                            for initiators in initiators_list:
-                                initiators_durable_id = initiators.get(
-                                    'native_storage_host_initiator_id')
-                                if initiators_durable_id == mapped_id:
-                                    native_storage_host_id = initiators.get(
-                                        'native_storage_host_id')
+                    native_host_group_name = 'native_storage_host_id'
```
Comment (ghca-cxl) [inline delfin/drivers/hpe/hpe_msa/ssh_handler.py:749]: It can be combined into one statement

**Pull request merged**